src/App.tsx

```diff
@@ -1,223 +1,196 @@
+      setDeletedTodoIds((currentIDs) => [...currentIDs, todoId]);
+
+      await deleteTodo(todoId);
+      setTodos(currentTodos => {
+        return currentTodos.filter(todo => todoId !== todo.id);
+      });
+
+      return 1;
+    } catch {
+      setError(ErrorTypes.DELETE);
+      setIsErrorHidden(false);
+
+      return 0;
+    } finally {
+      setDeletedTodoIds([]);
+    }
+  };
+
+  useEffect(() => {
+    getUserTodos();
   }, []);
 
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
-
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
+  const filterBySelect = useCallback((
+    todosFromServer: Todo[],
+    option: string,
+  ) => {
+    return todosFromServer.filter(todo => {
+      switch (option) {
+        case FilterOptions.ACTIVE:
+          return todo.completed === false;
+
+        case FilterOptions.COMPLETED:
+          return todo.completed === true;
+
+        case FilterOptions.ALL:
+        default:
+          return true;
+      }
+    });
+  }, []);
 
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const visibleTodos = useMemo(() => {
+    return filterBySelect(todos, filtredBy);
+  }, [todos, filtredBy]);
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+  const amountOfActiveTodos = useMemo(() => {
+    return filterBySelect(todos, FilterOptions.ACTIVE).length;
+  }, [todos]);
```

Comment:
why make another call of a function?
```suggestion
  const amountOfActiveTodos = visibleTodos.length
```

Reply:
yes, I've already fixed this in the third part)